hdat/suite.py

```diff
@@ -71,24 +74,20 @@ def collect_suites(directory):
 
 
 def _collect_suite_classes(directory):
-    suites_filename = find_here_or_in_parents(directory, '.hdattsuites')
-    if suites_filename is None:
-        raise AbortError('Unable to locate a ".hdattsuites" file')
+    hdat_module_suffix = '_hdat.py'
+    hdat_suite_class = Suite
 
     suite_classes = []
-    with open(suites_filename, 'r') as suites_file:
-        for line in suites_file:
-            class_location = line.strip()
-            try:
-                test_suite_class = pydoc.locate(class_location)
-            except pydoc.ErrorDuringImport as e:
-                print_error(e)
-                test_suite_class = None
-
-            if test_suite_class is None:
-                msg = 'Unable to import test suite "{}"'
-                raise AbortError(msg.format(class_location))
-            else:
-                suite_classes.append(test_suite_class)
-
+    for root, dirs, files in os.walk(directory, topdown=True):
+        # prevent os.walk from going into hidden dirs
+        dirs = [subdir for subdir in dirs if not subdir.startswith('.')]
+        for filename in files:
+            if filename.endswith(hdat_module_suffix):
+                module_name = filename.strip(".py")
```

Comment:
It may be nice to split the inside of this loop into a separate function.

Also, it seems that we could run into issues if we rely on `sys.path.append` to import modules.  For example, what if there are modules named the same thing in different folders?  It seems like the first one will be imported twice.

Here is an experiment demonstrating the issue:

```bash
~/Desktop/test$ ls -laR
total 8
drwxr-xr-x  5 johndavidgiese  staff  170 Feb 20 10:23 .
drwxr-xr-x@ 9 johndavidgiese  staff  306 Feb 20 10:22 ..
drwxr-xr-x  4 johndavidgiese  staff  136 Feb 20 10:24 a
drwxr-xr-x  3 johndavidgiese  staff  102 Feb 20 10:22 b
-rw-r--r--  1 johndavidgiese  staff  124 Feb 20 10:23 main.py

./a:
total 16
drwxr-xr-x  4 johndavidgiese  staff  136 Feb 20 10:24 .
drwxr-xr-x  5 johndavidgiese  staff  170 Feb 20 10:23 ..
-rw-r--r--  1 johndavidgiese  staff   17 Feb 20 10:22 hello.py
-rw-r--r--  1 johndavidgiese  staff  116 Feb 20 10:24 hello.pyc

./b:
total 8
drwxr-xr-x  3 johndavidgiese  staff  102 Feb 20 10:22 .
drwxr-xr-x  5 johndavidgiese  staff  170 Feb 20 10:23 ..
-rw-r--r--  1 johndavidgiese  staff   17 Feb 20 10:22 hello.py
~/Desktop/test$ cat main.py 
import sys
import importlib

sys.path.append('./a')
sys.path.append('./b')
print(sys.path)
importlib.import_module('hello')
~/Desktop/test$ cat a/hello.py
print('hello a')
~/Desktop/test$ cat b/hello.py 
print('hello b')
~/Desktop/test$ python main.py 
['/Users/johndavidgiese/Desktop/test', '/usr/local/Cellar/python/2.7.12/Frameworks/Python.framework/Versions/2.7/lib/python27.zip', '/usr/local/Cellar/python/2.7.12/Frameworks/Python.framework/Versions/2.7/lib/python2.7', '/usr/local/Cellar/python/2.7.12/Frameworks/Python.framework/Versions/2.7/lib/python2.7/plat-darwin', '/usr/local/Cellar/python/2.7.12/Frameworks/Python.framework/Versions/2.7/lib/python2.7/plat-mac', '/usr/local/Cellar/python/2.7.12/Frameworks/Python.framework/Versions/2.7/lib/python2.7/plat-mac/lib-scriptpackages', '/usr/local/Cellar/python/2.7.12/Frameworks/Python.framework/Versions/2.7/lib/python2.7/lib-tk', '/usr/local/Cellar/python/2.7.12/Frameworks/Python.framework/Versions/2.7/lib/python2.7/lib-old', '/usr/local/Cellar/python/2.7.12/Frameworks/Python.framework/Versions/2.7/lib/python2.7/lib-dynload', '/usr/local/lib/python2.7/site-packages', '/Library/Python/2.7/site-packages', './a', './b']
hello a
```

Maybe we can find a way to import the modules directly?

Also, it seems like mutating the `sys.path` could cause problems in user's code if they do any sort of dynamic imports.

Reply:
The two names issue comes up if there are two hdat suite containing modules with the same module name under the same directory. This is something that we can decide to support or not; for example, `nose` supports this by importing a module, running its tests, and then resetting the path, while `nose2` explicitly does not support this. More details about this at https://nose2.readthedocs.io/en/latest/differences.html#test-discovery-and-loading

Messing with `sys.path` felt kind of flaky to me as well, but it has to happen somewhere in order to import arbitrary modules in unknown directories. For example, pytest uses `pyimport` in the `py` module which ultimately modifies `sys.path`: https://github.com/pytest-dev/py/blob/master/py/_path/local.py#L638 specifically the `modname is None` case. Nose2 does something similar at https://github.com/nose-devs/nose2/blob/master/nose2/util.py#L216

Reply:
> The two names issue comes up if there are two hdat suite containing modules with the same module name under the same directory.

I thought the issue would come up if there are two hdat-suite-containing modules with the same name in any directories.

E.g. if you had:

```
project/a/b/c/my_hdat.py
project/d/e/f/my_hdat.py
```
And the sys.path was `['project/a/b/c', 'project/d/e/f']` then (I think) you would not be able to import the second file.

Reply:
Right, I meant the directory `hdat` was run in. I can look into scoping the modules like `a.b.c.my_hdat` and `d.e.f.my_hdat`. The `__package__` attribute also might need to be set to make relative imports in the user code work.

Reply:
Ahh, I see.

I seems that scoping the modules like `a.b.c.my_hdat` and `d.e.f.my_hdat` will:

1. It will let us support multiple hdat suites with the same name
2. We will at most need to add a single path to the `sys.path`